std/log/handlers.ts

```diff
@@ -85,33 +87,121 @@ export class ConsoleHandler extends BaseHandler {
 
 export abstract class WriterHandler extends BaseHandler {
   protected _writer!: Writer;
-  private _encoder = new TextEncoder();
-
+  #encoder = new TextEncoder();
+
+  /** @deprecated - `this._writer.write()` is an async call.  Under heavy load
+   * testing has shown a significant loss of log messages being written.  Also,
+   * if the process terminates while writing a log, that log message is also
+   * lost.  It is recommended that handlers override this method with a
+   * synchronous implementation (see FileHandler).
+   */
   log(msg: string): void {
-    this._writer.write(this._encoder.encode(msg + "\n"));
+    this._writer.write(this.#encoder.encode(msg + "\n"));
   }
 }
 
 interface FileHandlerOptions extends HandlerOptions {
   filename: string;
+  mode?: LogMode;
 }
 
 export class FileHandler extends WriterHandler {
-  private _file!: File;
-  private _filename: string;
+  protected _file!: File;
+  protected _filename: string;
+  protected _mode: LogMode;
+  #encoder = new TextEncoder();
 
   constructor(levelName: string, options: FileHandlerOptions) {
     super(levelName, options);
     this._filename = options.filename;
+    // default to append mode, write only
+    this._mode = options.mode ? options.mode : "a";
   }
 
   async setup(): Promise<void> {
-    // open file in append mode - write only
-    this._file = await open(this._filename, "a");
+    this._file = await open(this._filename, this._mode);
     this._writer = this._file;
   }
 
+  log(msg: string): void {
```

Comment:
Just an idea - support both `log` and `logSync` methods?

Reply:
Yes, I considered this too.  This definitely needs more thought though as I think the async implementation isn't straightforward, at least to get something robust which can handle high throughput.  Something for another PR I think.